drivers/flash/spi_nor.c

```diff
@@ -564,21 +556,19 @@ static void acquire_device(const struct device *dev)
 		k_sem_take(&driver_data->sem, K_FOREVER);
 	}
 
-	if (IS_ENABLED(CONFIG_SPI_NOR_IDLE_IN_DPD)) {
-		exit_dpd(dev);
-	}
+	/* Ensure device is out of DPD */
+	(void)pm_device_runtime_get(dev);
 }
 
-/* Everything necessary to release access to the device.
- *
- * This means (optionally) putting the device into deep power-down
- * mode, and releasing the lock.
- */
+#ifndef CONFIG_SPI_NOR_DEEP_POWER_DOWN_INACTIVITY_TIMEOUT_MS
+#define CONFIG_SPI_NOR_DEEP_POWER_DOWN_INACTIVITY_TIMEOUT_MS 0
```

Comment:
Do not redefine CONFIG_ when not available, use local identifier.
All c preprocessor definitions should happen on the top of the unit.

Reply:
Actually it should be impossible to have this undefined, is this even needed?

Reply:
The symbol `CONFIG_SPI_NOR_DEEP_POWER_DOWN_INACTIVITY_TIMEOUT_MS` depends on `PM_DEVICE_RUNTIME`, so it is undefined when built without PM.
Fixed the initial request